### PR TITLE
AF-2011: Redeploy does not abort running process instances when their process definition is deleted

### PR DESCRIPTION
**Pull request by pefernan**
@mswiderski  @domhanak would you mind take a look?

This is part of an ensemble, please merge with:
https://github.com/kiegroup/kie-wb-common/pull/2681
https://github.com/kiegroup/droolsjbpm-integration/pull/1839

### CODE REVIEW

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java [hunk 1]
```diff
@@ -953,7 +955,17 @@ private Message updateKieContainerToVersion(KieContainerInstanceImpl kci, Releas
         return response;
     }
 
-    private List<Message> updateExtensions(KieContainerInstanceImpl kci, ReleaseId releaseId, List<Message> messages, boolean resetBeforeUpdate) {
+    private void prepareUpdateExtensions(KieContainerInstanceImpl kci, ReleaseId releaseId, List<Message> messages, boolean resetBeforeUpdate) {
+        Map<String, Object> parameters = getReleaseUpdateParameters(releaseId, messages, resetBeforeUpdate);
+
+        // once the upgrade was successful, notify all extensions so they can be upgraded (if needed)
```
Comment (mswiderski) [inline kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java]: This comment seems to be in wrong place

Comment (pefernan) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java]: @mswiderski you are right!

**Comment by domhanak** [comment]
Roses are red,
violets are blue,
Jenkins execute full downstream build!

**Comment by pefernan** [comment]
jenkins execute full downstream build

**Comment by pefernan** [comment]
jenkins execute full downstream build

**Comment by pefernan** [comment]
jenkins retest this

**Pull request merged**